### ISSUE TITLE
`prefer-lookaround` Not checking  groups contains negate CharacterClass

### ISSUE THREAD

**Issue by fisker** [ota-meshi/eslint-plugin-regexp#484]
**Information:**
- ESLint version: 8.27.0
- `eslint-plugin-regexp` version: 7.31.10

**Description**

```js
'ab'.replaceAll(/a([b])/g, "$1");
'ac'.replaceAll(/a([^b])/g, "$1");
```

https://ota-meshi.github.io/eslint-plugin-regexp/playground/#eJyrVkrOT0lVslJST0xS1ytKLchJTE51zMnR0E/UiE6K1dRP11GIUVIxjFHStI7JU09MxlAUh6ZKSUepqDQntVjJqlqpKDU9taJAv6AoNS21SDcnPz87sSi/NC9FycqothYA9c4mgQ==

The first is reported. 

The second is not, it should be able to fix to

```js
'ac'.replaceAll(/a(?=[^b])/g, "");
```

**Comment by ota-meshi** [comment]
Thank you for posting issue.

But, it's an intentional behavior.
Replacing it with lookahead has the side effect of not consuming any characters.

```js
'aaaaa'.replaceAll(/a([^b])/g, ",$1");
// ',a,aa'
'aaaaa'.replaceAll(/a(?=[^b])/g, ",");
// ',,,,a'
```

The following negated character classes are reported because they have no side effects when replaced with lookahead.

```js
'ac'.replaceAll(/a([^a])/g, "$1");
```

**Comment by ota-meshi** [comment]
Thank you for posting issue.

But, it's an intentional behavior.
Replacing it with lookahead has the side effect of not consuming any characters.

```js
'aaaaa'.replaceAll(/a([^b])/g, ",$1");
// ',a,aa'
'aaaaa'.replaceAll(/a(?=[^b])/g, ",");
// ',,,,a'
```

The following negated character classes are reported because they have no side effects when replaced with lookahead.

```js
'ac'.replaceAll(/a([^a])/g, "$1");
```

**Comment by fisker** [comment]
Thanks for the information.
